Automattic/newspack-newsletters
# fix: migrate newsletters post status

### All Submissions:






### Changes proposed in this Pull Request:

Should be reviewed and merged after #824 

Removes the custom query hooks and implements a gradual/progressive migration of newsletters' post status to `private` for posts without `is_public` meta. When `the_post` fires on the frontend, a refresh should take place after the post update.

### How to test the changes in this Pull Request:

1. Check out this branch and send a newsletter that won't be public
2. Manually change the `post_status` to `publish` on the database
3. While logged out, search for a term that matches the title/text on the newsletter
4. Observe the database is updated and the page results does not include the private newsletter

### Other information:
